asreview/webapp/auth.py

```diff
@@ -0,0 +1,137 @@
+# Copyright 2019-2022 The ASReview Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import datetime
+from pathlib import Path
+
+from flask import Blueprint
+from flask import jsonify
+from flask import request
+from flask_cors import CORS
+from flask_login import current_user, login_user, logout_user
+from sqlalchemy.exc import SQLAlchemyError
+
+from asreview.webapp import DB
+from asreview.webapp.authentication.login_required import asreview_login_required
+from asreview.webapp.authentication.models import User
+
+bp = Blueprint('auth', __name__, url_prefix='/auth')
+CORS(
+    bp,
+    resources={r"*": {"origins": "http://localhost:3000"}},
+    supports_credentials=True,
+)
+
+
+@bp.route('/signin', methods=["POST"])
+def signin():
+    print(request.form)
+    email = request.form.get('email').strip()
+    password = request.form.get('password')
+
+    # get the user
+    user = User.query.filter(User.email == email).one_or_none()
+    # if the user exist proceed and verify
+    if not user:
+        result = (404, {'message': f'User account {email} does not exist.'})
+    else:
+        # verify password
+        if user.verify_password(password):
```

Comment:
Maybe we can make this a bit more modular, so that it is easier to implement different methods of authenticating. We could make a function `authenticate_user(user, request)` that authenticates a user based on a request, which in the default case would be `def authenticate_user(user, request): return check_password_hash(user.hashed_password, request.form.get('password'))`. In this way, if someone wants to implement a different method of authenticating, they would only need to give a different implementation of `authenticate_user`. Then we could allow extensions of the ASReview authentication in the same way that we allow extension of models.